src/components/Map/index.tsx

```diff
@@ -231,11 +232,9 @@ const Map = () => {
   }, []);
 
   const handleClickForeCastButton = () => {
-    navigate(
-      `${ROUTE.DUST_FORECAST}?sido=${encodeURIComponent(
-        currentSido
-      )}&city=${encodeURIComponent(city)}`
-    );
+    SIDO_NAMES.includes(city)
+      ? navigate(`${ROUTE.RANKING}/${city}`)
+      : navigate(`${ROUTE.DUST_FORECAST}?sido=${currentSido}&city=${city}`);
```

Comment:
Map 컴포넌트 내부의 상태 city를 sido로 변경하면 더 명확할 것 같네요. 이미 useMap 훅에서 currentCity가 아닌 currentSido로 관리되고 있어서 통일하면 좋을 것 같아요.
```suggestion
    const [sido, setSido] = useState('시도명');

    SIDO_NAMES.includes(sido)
      ? navigate(`${ROUTE.RANKING}/${sido}`)
      : navigate(`${ROUTE.DUST_FORECAST}?sido=${currentSido}&city=${city}`);
```

추가적으로 Map 컴포넌트를 보다가 코멘트 남겨요.
> 전체적으로 Map 컴포넌트를 추상화하면 좋을 것 같아요.
1. makeMarkerTemplate 함수는 컴포넌트 내부에 없어도 될 것 같아요.
2. 중복 로직들이 너무 많아요. 유틸 함수나 훅으로 분리해도 좋을 것 같아요.
3. Map 컴포넌트 내부의 Modal 부분도 분리하면 좋을 것 같네요.
4. useEffect가 너무 많아서 각각 어떤 일을 하는지 직관적으로 파악하기 힘들어요. 내부에 주석을 이용하거나, 복잡한 조건은 따로 명시적인 함수명으로 분리해도 좋을 것 같아요.

Reply:
`city`는 본래 sido 보다 작은 단위인 동네들을 나타내기 위한 변수였어요. 그런데 지도에서 `sido` 단위 마커를 클릭하게 되면 상세 정보 페이지로 이동할 수 없기 때문에 지도에서 `sido`단위 마크가 표기될 때 해당 마커를 클릭하면 예외적으로 동네 단위가 아니라 `sido`단위를 나타내게 됩니다.
물론 본래 목적이 동네 단위를 나타내기 때문에 위의 케이스가 아니면 동네 단위 정보만 담고 있고 활용됩니다. 위에서 언급한 경우를 처리하면서 이름은 그대로 사용하다 보니 혼선이 생긴 것같아요. 
`city`에서 `sido`로 바꾸게되면 다른 함수에서 혼선이 생길 수 있다고 생각되요. 좀 더 좋은 방법을 생각해 볼게요.

Map 컴포넌트 리팩터링은 인지하고 있어요.  남겨주신 내용도 참고하도록 할게요! 감사합니다 😀

Reply:
아하...! 그럼 머지하고 이후에 남은 작업 진행해주세용~! 👍👍👍